earth_enterprise/src/common/tests/RunAllTests.pl

```diff
@@ -17,12 +17,24 @@
 
 use FindBin qw($Bin $Script);
 use File::Basename;
+use Cwd;
+use Cwd 'abs_path';
+
+# This script should be run from NATIVE-*-x86_64/bin/tests
+my $workingDir = abs_path(cwd());
+my $scriptDir = dirname(abs_path($0));
+my $notInBuildDir = (index($scriptDir, "NATIVE") == -1);
+if ($workingDir ne $scriptDir || $notInBuildDir) {
```

Comment:
I agree with this. It would be more user friendly in my opinion to just cd into the proper directory.